src/test/run-pass/catch-expr.rs

```diff
@@ -10,8 +10,10 @@
 
 #![feature(catch_expr)]
 
+struct catch {}
+
 pub fn main() {
-    let catch_result = catch {
+    let catch_result = do catch {
```

Comment:
seeing as how there is special code around restrictions, shouldn't we have a test like `while do catch { ... }` and show that it results in an error?